lib/poison/decoder.ex

```diff
@@ -40,9 +40,13 @@ defmodule Poison.Decode do
   end
 
   defp transform_struct(value, _keys, as, options) do
-    kv = for k <- Map.keys(Map.from_struct(as.__struct__)) do
-      {k, Map.get(value, Atom.to_string(k))}
-    end
+    kv = Map.keys(Map.from_struct(as.__struct__))
+          |> Enum.reduce([], fn key, acc ->
+               case Map.get(value, Atom.to_string(key)) do
+                 nil -> acc
+                 value -> [{key, value} | acc]
+               end
+             end)
```

Comment:
`value` should have strings instead of atoms for keys, e.g. `%{"name" => "john"}` -- I don't think `struct/2` handles mapping keys from atom to string?

```elixir
iex> struct(Person, %{"name" => "john"})
%Person{age: 42, name: nil}
```
Thanks!